src/leveled_cdb.erl

```diff
@@ -762,26 +769,24 @@ delete_pending({key_check, Key}, _From, State) ->
 delete_pending(timeout, State=#state{delete_point=ManSQN}) when ManSQN > 0 ->
     case is_process_alive(State#state.inker) of
         true ->
-            case leveled_inker:ink_confirmdelete(State#state.inker, ManSQN) of
-                true ->
-                    leveled_log:log("CDB04", [State#state.filename, ManSQN]),
-                    close_pendingdelete(State#state.handle, 
-                                        State#state.filename, 
-                                        State#state.waste_path),
-                    {stop, normal, State};
-                false ->
-                    {next_state,
-                        delete_pending,
-                        State,
-                        ?DELETE_TIMEOUT}
-            end;
+            ok = 
```

Comment:
If the process is alive, you may send the async message for each new timeout. That is, if DELETE_TIMEOUT is 1 second and the inker is very busy, then you may send 10 messages in 10 seconds.
This would be impossible before, since you wait for the inker and if the inker is busy, you would timeout the actual call.

Reply:
Yes, and any one of those requests leading to a confirmation will cause the leveled_cdb file to close.